backend/config/storage.go

```diff
@@ -0,0 +1,42 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type StorageConfig struct {
+	Path string `yaml:"path"`
+}
+
+func validateStorageConfig(cfg *Config) error {
+	if cfg.Storage == nil {
+		cfg.Storage = &StorageConfig{
+			Path: DefaultStoragePath,
+		}
+	}
+
+	if len(cfg.Storage.Path) == 0 {
+		cfg.Storage.Path = DefaultStoragePath
+	}
+
+	info, err := os.Stat(cfg.Storage.Path)
+	if os.IsNotExist(err) {
+		// if it is a local path, create it
+		if !strings.HasPrefix(cfg.Storage.Path, "/") &&
+		!strings.HasPrefix(cfg.Storage.Path, "../") &&
+		!strings.HasPrefix(cfg.Storage.Path, "~/") {
```

Comment:
This check is good! It's [agreed upon](https://github.com/golang/go/issues/4140) to not bother using `~` as the home directory. If the user wants to use the home directory, they should use `$HOME` which is expanded before in the `os. ExpandEnv` invocation. Made a note to include this in future documentation ✅